### PR TITLE
[PROTOCOL-307, PROTOCOL-295] Implement Integration Test Scenarios

### PR DESCRIPTION
**Pull request by salazarguille**
It includes:

- Create generic functions (actions) to call contracts.
- Implement some integration tests as examples.

image

### CODE REVIEW

File: test-integration/initializers/index.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const _2_add_signers = require('./2_add_signers');
 
 const initializersObject = {
     'init-1-set-settings': _1_set_settings,
-    'init-2-add-signers': _2_add_signers,
+    // 'init-2-add-signers': _2_add_signers,
```
Comment (salazarguille) [inline test-integration/initializers/index.js:7]: Why is it commented?

File: test-integration/main.js [hunk 2]
```diff
@@ -22,7 +25,9 @@ const executeTestFunction = async (testFunctionObject, testContext) => {
 };
 
 module.exports = async (callback) => {
+    const network = processArgs.getValue('network');
     const revertBlockchain = processArgs.getValue('revert', false);
+    // const revertTestBlockchain = processArgs.getValue('revertTest', true);
```
Comment (salazarguille) [inline test-integration/main.js:30]: Why is it commented?

File: test-integration/main.js [hunk 3]
```diff
@@ -56,7 +68,14 @@ module.exports = async (callback) => {
                 for (const testObject of testObjects) {
                     try {
                         printSeparatorLine();
+                        // let testSnapshotId;
+                        // if(revertTestBlockchain) {
```
Comment (salazarguille) [inline test-integration/main.js:72]: Why is it commented?

File: test-integration/main.js [hunk 4]
```diff
@@ -56,7 +68,14 @@ module.exports = async (callback) => {
...
+                        //     testSnapshotId = await timer.takeSnapshot();
+                        // }
                         await executeTestFunction(testObject, testContext);
+                        // if(revertTestBlockchain) {
```
Comment (salazarguille) [inline test-integration/main.js:76]: Why is it commented?

File: test-integration/utils/Swapper.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+class Swapper {
+  constructor(web3, deployedArtifact) {
+    this.web3 = web3
+    this.artifact = deployedArtifact
+  }
+
+  static async init(web3, deployedArtifact, funderTxConfig) {
+    const uniswap = new Swapper(web3, deployedArtifact)
+
+    // fund contract with 500 ETH
```
Comment (salazarguille) [inline test-integration/utils/Swapper.js:10]: It doesn't work on ganache-cli. By default each account has 100 eth in ganache.

File: test-integration/utils/actions/escrow.js [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    teller.escrowDapp(logicNames.Uniswap)
+  );
+  const dappData = {
+    location: uniswapDappAddress,
+    data: uniswapDapp.methods.swap(path, sourceAmount, minDestination).encodeABI()
+  };
+  const swapResult = await escrow.callDapp(dappData, txConfig);
+
+  // const sourceTokenAddress = path[0];
+  // const destinationTokenAddress = path[path.length - 1];
```
Comment (salazarguille) [inline test-integration/utils/actions/escrow.js:80]: Why is it commented?

**Review by passabilities — APPROVED** [review]

**Review by Juan-cc — APPROVED** [review]

**Pull request merged**